ClassNotFoundException: org.openhab.habdroid.model.ServerProperties

<!-- Please search the issue, if there is one with your issue -->

### Actual behaviour
Open prefs and get the following stack trace in the log, but the app doesn't crash:
````
2022-12-09 22:24:45.241   934-949   Parcel                  pid-934                              E  Class not found when unmarshalling: org.openhab.habdroid.model.ServerProperties
                                                                                                    java.lang.ClassNotFoundException: org.openhab.habdroid.model.ServerProperties
                                                                                                    	at java.lang.Class.classForName(Native Method)
                                                                                                    	at java.lang.Class.forName(Class.java:306)
                                                                                                    	at java.lang.Class.forName(Class.java:270)
                                                                                                    	at android.os.Parcel.readParcelableCreator(Parcel.java:2316)
                                                                                                    	at android.os.Parcel.readParcelable(Parcel.java:2280)
                                                                                                    	at android.os.Parcel.readValue(Parcel.java:2187)
                                                                                                    	at android.os.Parcel.readArrayMapInternal(Parcel.java:2520)
                                                                                                    	at android.os.BaseBundle.unparcel(BaseBundle.java:224)
                                                                                                    	at android.os.BaseBundle.getString(BaseBundle.java:922)
                                                                                                    	at com.amazon.startingwindow.CustomStartingWindow.updateURIOverrides(CustomStartingWindow.java:86)
                                                                                                    	at com.amazon.startingwindow.CustomStartingWindow.setStartingWindowURIs(CustomStartingWindow.java:121)
                                                                                                    	at com.android.server.am.StartingWindowActivityStackCallback.onAppStartingPreview(StartingWindowActivityStackCallback.java:21)
                                                                                                    	at com.android.server.am.VendorActivityStackCallback.callOnAppStartingPreview(VendorActivityStackCallback.java:44)
                                                                                                    	at com.android.server.am.ActivityStack.startActivityLocked(ActivityStack.java:2233)
                                                                                                    	at com.android.server.am.ActivityStackSupervisor.startActivityUncheckedLocked(ActivityStackSupervisor.java:2395)
                                                                                                    	at com.android.server.am.ActivityStackSupervisor.startActivityLocked(ActivityStackSupervisor.java:1666)
                                                                                                    	at com.android.server.am.ActivityStackSupervisor.startActivityMayWait(ActivityStackSupervisor.java:1022)
                                                                                                    	at com.android.server.am.ActivityStackSupervisor.startActivityMayWait(ActivityStackSupervisor.java:891)
                                                                                                    	at com.android.server.am.ActivityManagerService.startActivityAsUser(ActivityManagerService.java:3736)
                                                                                                    	at com.android.server.am.ActivityManagerService.startActivity(ActivityManagerService.java:3694)
                                                                                                    	at android.app.ActivityManagerNative.onTransact(ActivityManagerNative.java:139)
                                                                                                    	at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:2438)
                                                                                                    	at android.os.Binder.execTransact(Binder.java:446)
                                                                                                    Caused by: java.lang.ClassNotFoundException: Didn't find class "org.openhab.habdroid.model.ServerProperties" on path: DexPathList[[directory "."],nativeLibraryDirectories=[/vendor/lib, /system/lib]]
                                                                                                    	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:56)
                                                                                                    	at java.lang.ClassLoader.loadClass(ClassLoader.java:511)
                                                                                                    	at java.lang.ClassLoader.loadClass(ClassLoader.java:469)
                                                                                                    	at java.lang.Class.classForName(Native Method) 
                                                                                                    	at java.lang.Class.forName(Class.java:306) 
                                                                                                    	at java.lang.Class.forName(Class.java:270) 
                                                                                                    	at android.os.Parcel.readParcelableCreator(Parcel.java:2316) 
                                                                                                    	at android.os.Parcel.readParcelable(Parcel.java:2280) 
                                                                                                    	at android.os.Parcel.readValue(Parcel.java:2187) 
                                                                                                    	at android.os.Parcel.readArrayMapInternal(Parcel.java:2520) 
                                                                                                    	at android.os.BaseBundle.unparcel(BaseBundle.java:224) 
                                                                                                    	at android.os.BaseBundle.getString(BaseBundle.java:922) 
                                                                                                    	at com.amazon.startingwindow.CustomStartingWindow.updateURIOverrides(CustomStartingWindow.java:86) 
                                                                                                    	at com.amazon.startingwindow.CustomStartingWindow.setStartingWindowURIs(CustomStartingWindow.java:121) 
                                                                                                    	at com.android.server.am.StartingWindowActivityStackCallback.onAppStartingPreview(StartingWindowActivityStackCallback.java:21) 
                                                                                                    	at com.android.server.am.VendorActivityStackCallback.callOnAppStartingPreview(VendorActivityStackCallback.java:44) 
                                                                                                    	at com.android.server.am.ActivityStack.startActivityLocked(ActivityStack.java:2233) 
                                                                                                    	at com.android.server.am.ActivityStackSupervisor.startActivityUncheckedLocked(ActivityStackSupervisor.java:2395) 
                                                                                                    	at com.android.server.am.ActivityStackSupervisor.startActivityLocked(ActivityStackSupervisor.java:1666) 
                                                                                                    	at com.android.server.am.ActivityStackSupervisor.startActivityMayWait(ActivityStackSupervisor.java:1022) 
                                                                                                    	at com.android.server.am.ActivityStackSupervisor.startActivityMayWait(ActivityStackSupervisor.java:891) 
                                                                                                    	at com.android.server.am.ActivityManagerService.startActivityAsUser(ActivityManagerService.java:3736) 
                                                                                                    	at com.android.server.am.ActivityManagerService.startActivity(ActivityManagerService.java:3694) 
                                                                                                    	at android.app.ActivityManagerNative.onTransact(ActivityManagerNative.java:139) 
                                                                                                    	at com.android.server.am.ActivityManagerService.onTransact(ActivityManagerService.java:2438) 
                                                                                                    	at android.os.Binder.execTransact(Binder.java:446) 
                                                                                                    	Suppressed: java.lang.ClassNotFoundException: org.openhab.habdroid.model.ServerProperties
                                                                                                    		at java.lang.Class.classForName(Native Method)
                                                                                                    		at java.lang.BootClassLoader.findClass(ClassLoader.java:781)
                                                                                                    		at java.lang.BootClassLoader.loadClass(ClassLoader.java:841)
                                                                                                    		at java.lang.ClassLoader.loadClass(ClassLoader.java:504)
                                                                                                    		... 24 more
                                                                                                    	Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack available
````


### Expected behaviour
No stack trace in log.

 
### Environment data
#### Client
* Android version: 5.1.1
* Device model: Amazon Fire Tablet
* App version <!-- Can be found under "About" -->:  2.23.6

--- a/mobile/src/main/java/org/openhab/habdroid/ui/preference/fragments/MainSettingsFragment.kt
+++ b/mobile/src/main/java/org/openhab/habdroid/ui/preference/fragments/MainSettingsFragment.kt
@@ -25,7 +25,6 @@ import android.os.Bundle
 import android.provider.Settings
 import android.util.Log
 import androidx.activity.result.contract.ActivityResultContracts
-import androidx.annotation.ColorInt
 import androidx.annotation.RequiresApi
 import androidx.annotation.StringRes
 import androidx.appcompat.app.AppCompatDelegate

--- a/mobile/src/main/java/org/openhab/habdroid/util/ExtensionFuncs.kt
+++ b/mobile/src/main/java/org/openhab/habdroid/util/ExtensionFuncs.kt
@@ -566,14 +566,20 @@ val PendingIntent_Mutable = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) 
     0
 }
 
-inline fun <reified T> Intent.parcelable(key: String): T? = when {
-    Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU -> getParcelableExtra(key, T::class.java)
-    else -> @Suppress("DEPRECATION") getParcelableExtra(key) as? T
+inline fun <reified T> Intent.parcelable(key: String): T? {
+    setExtrasClassLoader(T::class.java.classLoader)
+    return when {
+        Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU -> getParcelableExtra(key, T::class.java)
+        else -> @Suppress("DEPRECATION") getParcelableExtra(key) as? T
+    }
 }
 
-inline fun <reified T> Intent.parcelableArrayList(key: String): List<T>? = when {
-    Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU -> getParcelableArrayListExtra(key, T::class.java)
-    else -> @Suppress("DEPRECATION") getParcelableArrayListExtra(key)
+inline fun <reified T> Intent.parcelableArrayList(key: String): List<T>? {
+    setExtrasClassLoader(T::class.java.classLoader)
+    return when {
+        Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU -> getParcelableArrayListExtra(key, T::class.java)
+        else -> @Suppress("DEPRECATION") getParcelableArrayListExtra(key)
+    }
 }
 
 inline fun <reified T> Bundle.parcelable(key: String): T? = when {

